pkg/memcached/statefulset.go

```diff
@@ -57,7 +57,7 @@ func StatefulSet(m *memcachedv1.Memcached) *appsv1.StatefulSet {
 					Labels: ls,
 				},
 				Spec: corev1.PodSpec{
-					ServiceAccountName: "mariadb-operator-mariadb",
```

Comment:
This was a bug. An explicit dependency on the mariadb operator's service "service account"